src/string_with_vars.ml

```diff
@@ -47,62 +47,69 @@ module Token = struct
 end
 
 (* Remark: Consecutive [Text] items are concatenated. *)
-let rec of_tokens : Token.t list -> item list = function
+let rec of_tokens
+  : Loc.t -> Token.t list -> part list = fun loc -> function
   | [] -> []
   | Open a :: String s :: Close b :: rest when a = b ->
-    Var (a, s) :: of_tokens rest
+    let (name, payload) =
+      match String.lsplit2 s ~on:':' with
+      | None -> ("", s)
+      | Some v -> v
+    in
+    Var { loc
+        ; name
+        ; payload
+        ; syntax =
+            begin match a with
+            | Parens -> Dollar_paren
+            | Braces -> Dollar_brace
+            end
+        } :: of_tokens loc rest
   | token :: rest ->
     let s = Token.to_string token in
-    match of_tokens rest with
+    match of_tokens loc rest with
     | Text s' :: l -> Text (s ^ s') :: l
     | l -> Text s :: l
 
-let items_of_string s = of_tokens (Token.tokenise s)
+let items_of_string loc s = of_tokens loc (Token.tokenise s)
+
+let _t : Sexp.Of_sexp.ast -> t = function
```

Comment:
Is this used? You might want to use another name since it won't be caught by warning 32.